### DISCUSSION TITLE
Step output and quotes issue

### DISCUSSION THREAD

**Discussion by vitalyk-multinarity** [community/community#32012] [Actions] [answered]
I want to use JSON output from certain step in the next step. Unfortunately somehow double quotes disappear in the process.
Here are my test steps:

```
- name: Prep debug
       id: prep
       run: |
         export JSON_VALUE='{"id":12345,"node_id":"blah-blah"}'
         echo $JSON_VALUE | jq .node_id # works
         echo "::set-output name=json_output::$JSON_VALUE"
     - name: Debug
       run: |
          echo "${{ steps.prep.outputs.json_output}}"
          echo ${{ steps.prep.outputs.json_output}} | jq . # doesn't work
```

Seems very trivial - in my  JSON_VALUE I have correct JSON includes quotes, but after passing it via outputs to the next step I see a string without quotes: 

`{id:12345,node_id:blah-blah}`

Any ideas how to fix it?

**Comment by airtower-luna** [comment, marked as answer]
The GitHub Actions template syntax (`${{ ... }}`) is handled before Bash ever sees the script, and *literally* substituted with the value. So in your case Bash sees this, note that there is no variable:

```sh
echo "{"id":12345,"node_id":"blah-blah"}"
```

And then it interprets the quotes as Bash syntax, so they're gone from the output. What you should do is pass the value to the "Debug" step as an environment variable, e.g.:

```yaml
- name: Debug
  env:
    JSON: ${{ steps.prep.outputs.json_output }}
  run: |
    echo "$JSON"
```

That way Bash is aware of the variable and can handle it accordingly. If you don't need the value beyond the current job you could also skip the step output and [write the JSON directly to a variable in `$GITHUB_ENV`](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable).

**Reply by vitalyk-multinarity** [reply]
Many thanks, you saved my day!
